scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -212,20 +232,15 @@ private int startTasks(Policy currentPolicy, Map<JobId, JobDescriptor> jobMap, M
             //if there is no free resources, stop it right now without starting any task
             if (freeResources.isEmpty()) {
 
-                final int eligibleByPolicy = fullListOfTaskRetrievedFromPolicy.stream()
-                                                                              .mapToInt(TaskDescriptor::getNumberOfNodesNeeded)
-                                                                              .sum();
-                // eligible by policy
-                getRMProxiesManager().getRmProxy().setPendingTasksCount(eligibleByPolicy);
-
+                updateStatsAboutPendingEligibleAndNeededNodes(fullListOfTaskRetrievedFromPolicy);
                 return 0;
             }
 
             schedulingMainLoopTimingLogger.end("getOrderedTasks");
 
             //if there is no task to scheduled, return without starting any task
             if (fullListOfTaskRetrievedFromPolicy == null || fullListOfTaskRetrievedFromPolicy.isEmpty()) {
-                getRMProxiesManager().getRmProxy().setPendingTasksCount(0);
+                updateStatsAboutPendingEligibleAndNeededNodes();
```

Comment:
before it was pending task set to 0, now it will update it according to needed nodes. Could you explain this point?